File: app/src/main/java/mikazuki/android/app/feelingmatch/view/activity/ResultActivity.java
package mikazuki.android.app.feelingmatch.view.activity;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.widget.TextView;

import com.annimon.stream.Stream;
import com.google.common.collect.Maps;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import butterknife.Bind;
import butterknife.ButterKnife;
import io.realm.Realm;
import io.realm.RealmConfiguration;
import mikazuki.android.app.feelingmatch.R;
import mikazuki.android.app.feelingmatch.model.Match;
import mikazuki.android.app.feelingmatch.model.User;
import mikazuki.android.app.feelingmatch.view.customview.LikeButtonView;

/**
 * @author haijimakazuki
 */
public class ResultActivity extends BaseActivity {

    @Bind(R.id.result)
    TextView mResult;
    @Bind(R.id.heart)
    LikeButtonView mHeart;

    private Realm mRealm;
    private Match mMatch;
    private List<User> mMembers;
    private Number mMatchNum;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);
        ButterKnife.bind(this);


        if (getIntent().getExtras() == null) {
            mResult.setText("0組");
            return;
        }
        // Matchのload
        final long matchId = getIntent().getExtras().getLong("id");
        RealmConfiguration realmConfig = new RealmConfiguration.Builder(this).build();
        mRealm = Realm.getInstance(realmConfig);
        mMatch = mRealm.where(Match.class).equalTo("id", matchId).findFirst();
        mMembers = new ArrayList<>(mMatch.getMembers());

        Map<Long, User> boys = Maps.newHashMap();
        Map<Long, User> girls = Maps.newHashMap();
        Stream.of(mMembers).forEach(u -> (u.isBoy() ? boys : girls).put(u.getId(), u));
        mMatchNum = Stream.of(boys).filter(b -> girls.get(b.getValue().getFavoriteUserId()).getFavoriteUserId() == b.getKey()).count();
        mResult.setText(mMatchNum + "組");
    }

    @Override
    protected void onResume() {
        super.onResume();
        mHeart.animate(false);
        new Handler().postDelayed(() -> mHeart.animate(true), 200);
    }
}

File: app/src/main/java/mikazuki/android/app/feelingmatch/view/activity/ResultActivity.kt
package mikazuki.android.app.feelingmatch.view.activity

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import butterknife.ButterKnife
import butterknife.OnClick
import com.annimon.stream.Stream
import com.google.common.collect.Maps
import io.realm.Realm
import io.realm.RealmConfiguration
import kotlinx.android.synthetic.main.activity_result.*
import mikazuki.android.app.feelingmatch.R
import mikazuki.android.app.feelingmatch.model.Match
import mikazuki.android.app.feelingmatch.model.User
import java.util.*

/**
 * @author haijimakazuki
 */
class ResultActivity : BaseActivity() {


    private var mRealm: Realm? = null
    private var mMatch: Match? = null
    private var mMembers: List<User>? = null
    private var mMatchNum: Number? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)
        ButterKnife.bind(this)


        if (intent.extras == null) {
            result.text = "0組"
            return
        }
        // Matchのload
        val matchId = intent.extras.getLong("id")
        val realmConfig = RealmConfiguration.Builder(this).build()
        mRealm = Realm.getInstance(realmConfig)
        mMatch = mRealm!!.where(Match::class.java).equalTo("id", matchId).findFirst()
        mMembers = ArrayList(mMatch!!.members)

        val boys = Maps.newHashMap<Long, User>()
        val girls = Maps.newHashMap<Long, User>()
        Stream.of(mMembers).forEach { u -> (if (u.sex == 1) boys else girls).put(u.id, u) }
        mMatchNum = Stream.of(boys).filter { b -> girls[b.value.favoriteUserId]?.favoriteUserId == b.key }.count()
        result.text = "${mMatchNum ?: 0}組"
    }

    override fun onResume() {
        super.onResume()
        heart.animate(false)
        Handler().postDelayed({ heart.animate(true) }, 200)
    }

    @OnClick(R.id.go_top)
    fun goTopScreen(v: View) {
        startActivity(Intent(this, MainActivity::class.java))
    }
}
